src/Common/Infrastructure/RejectInvalidQueryParametersFilter.cs

```diff
@@ -20,10 +20,7 @@ public void OnResourceExecuting(ResourceExecutingContext context)
 
             var configuration = context.HttpContext.RequestServices.GetService<IConfiguration>();
 
-            var format =
-                context.GetValueFromHeader("format")
-                ?? context.GetValueFromRouteData("format")
-                ?? context.GetValueFromQueryString("format");
+            var format = context.DetermineFormatParameter();
 
             var acceptType = request.GetTypedHeaders().DetermineAcceptType(format, context.ActionDescriptor);
```

Comment:
I think you can pull this out of `format.ContentType` and if you add AcceptType to `ContentFormat` too (you already have it I think) that as well.